docs/helm/endpoints_api.md

```diff
@@ -0,0 +1,323 @@
+----
+  _Install Helm release_
+
+* **URL**
+
+    `/api/helm/release`
+
+* **Method:**
+
+  `POST`
+
+*  **Post Data Params**
+
+```
+ {
+   name: [string],
+   namespace: [string]
+   chart_url: [string]
+   values: map[string]interface{}
+  }
+```
+
+*  **Example Request**
+```
+    {
+    	"name": "test-helm-release",
+    	"namespace": "default",
+    	"chart_url": "https://github.com/akashshinde/console/raw/helm_endpoints/pkg/helm/testdata/influxdb-3.0.2.tgz"
+        "values": { "service": {"type": "ClusterIP"} }
+    }
+```
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
+  * JSON encoded [Release structure](https://github.com/helm/helm/blob/master/pkg/release/release.go#L22)
+
+* **Error Response:**
+
+  * **Code:** 400 BAD REQUEST <br />
+    **Content:** `{ error : "error message" }`
+
+
+**Uninstall Helm Release**
+----
+  _Uninstall Helm release_
+
+* **URL**
+
+    `/api/helm/release`
+
+* **Method:**
+
+  `DELETE`
+
+*  **URL Params**
+
+   `ns=[string]` - Namespace
+
+   `name=[string]` - Helm Release Name
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
+  * JSON encoded [UninstallReleaseResponse structure](https://github.com/helm/helm/blob/93137abbb4d391accd23dc774eb4d02a36d7a5f9/pkg/release/responses.go#L19)
+
+  * **Code:** 404 NOT FOUND <br />
+    **Content:** `{ error : "error message" }`
+
+* **Error Response:**
+
+  * **Code:** 502 BAD GATEWAY <br />
+    **Content:** `{ error : "error message" }`
+
+
+**Upgrade Helm Release**
+----
+  _Upgrade Helm release_
+
+* **URL**
+
+    `/api/helm/release`
+
+* **Method:**
+
+  `PUT`
+
+*  **Put Data Params**
+
+```
+ {
+   name: [string],
+   namespace: [string]
+   chart_url: [string]
+   values: map[string]interface{}
+  }
+```
+
+*  **Example Request**
+```
+    {
+    	"name": "test-helm-release",
+    	"namespace": "default",
+    	"chart_url": "https://github.com/akashshinde/console/raw/helm_endpoints/pkg/helm/testdata/influxdb-3.0.2.tgz"
+        "values": { "service": {"type": "ClusterIP"} }
+    }
+```
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
+  * JSON encoded [Release structure](https://github.com/helm/helm/blob/master/pkg/release/release.go#L22)
+
+* **Error Response:**
+
+  * **Code:** 502 BAD GATEWAY <br />
+    **Content:** `{ error : "error message" }`
+
+  * **Code:** 404 NOT FOUND <br />
+    **Content:** `{ error : "error message" }`
+
+**Rollback Helm Release**
+----
+  _Rollback Helm release_
+
+* **URL**
+
+    `/api/helm/release`
+
+* **Method:**
+
+  `PATCH`
+
+*  **Put Data Params**
+
+```
+ {
+   name: [string],
+   namespace: [string]
+   version: [int]
+  }
+```
+
+*  **Example Request**
+```
+    {
+    	"name": "test-helm-release",
+    	"namespace": "default",
+        "version": 1,
+    }
+```
+
+
+* **Success Response:**
+
+  * **Code:** 200 <br />
+  * JSON encoded [Release structure](https://github.com/helm/helm/blob/master/pkg/release/release.go#L22)
+
+* **Error Response:**
+
+  * **Code:** 502 BAD GATEWAY <br />
```

Comment:
same

Reply:
might be, but that is not in the focus of PR - we have just extracted the existing doc into a separate file. If we need to do something, let's t do that in a separate PR, because that would also require the change of endpoint logic, not just the doc.